backend/app/Services/FileParser.php

```diff
@@ -41,29 +42,39 @@ public function __construct(
      */
     public function fetch()
     {
-      $this->updateTimestamp();
-
       return json_decode(file_get_contents(base_path('tests/fixtures/fp/all.json')));
     }
 
     /**
      * Loop over all local files.
      *
      * @param $files
+     * @return \Illuminate\Http\JsonResponse
      */
     public function updateDatabase($files)
     {
+      DB::beginTransaction();
+
+      $this->updateTimestamp();
+
       foreach($files as $type => $items) {
         $this->itemCategory = $type;
 
         foreach($items as $item) {
-          $this->handleStatus($item);
+          try {
+            $this->handleStatus($item);
+          } catch(\Exception $e) {
+            return response()->json($e->getMessage(), 400);
```

Comment:
Instead of blank status codes you could use [http status constants](https://github.com/symfony/http-foundation/blob/master/Response.php).